src/logic/safe/store/reducer/gatewayTransactions.ts

```diff
@@ -118,68 +105,24 @@ export const gatewayTransactionsReducer = handleActions<GatewayTransactionsState
+
+    // Queue is overwritten completely on every update
+    // CGW sends a list of items where some items are LABELS (next and queued),
+    // some are CONFLICT_HEADERS (ignored),
+    // and some are actual TRANSACTIONS.
+    // We split the queue into next and queued by the index of the "Queued" LABEL.
+    // Then group TRANSACTIONS by their nonces.
     [ADD_QUEUED_TRANSACTIONS]: (state, action: Action<QueuedPayload>) => {
       const { chainId, safeAddress, values } = action.payload
-      let newNext = state[chainId]?.[safeAddress]?.queued?.next || {}
-      // We must clone as we delete transactions that move from queue to next
-      let newQueued = { ...(state[chainId]?.[safeAddress]?.queued?.queued || {}) }
-
-      let label: 'next' | 'queued' | undefined
-
-      values.forEach((value) => {
-        if (isLabel(value)) {
-          // we're assuming that the first page will always provide `next` and `queued` labels
-          label = value.label.toLowerCase() as 'next' | 'queued'
-          return
-        }
-
-        if (
-          // Conflict headers are not needed in the current implementation
-          isConflictHeader(value) ||
-          !isMultisigExecutionInfo(value.transaction.executionInfo)
-        ) {
-          return
-        }
-
-        const txNonce = value.transaction.executionInfo?.nonce
-
-        if (txNonce === undefined) {
-          console.warn('A transaction without nonce was provided by client-gateway:', JSON.stringify(value))
-          return
-        }
-
-        if (!label) {
-          const oldNext = state[chainId]?.[safeAddress]?.queued?.next
-          label = oldNext?.[txNonce] ? 'next' : 'queued'
-        }
-
-        const newTx = value.transaction
-
-        if (label === 'queued') {
-          if (newQueued?.[txNonce]) {
-            newQueued[txNonce] = getUpdatedQueueTxs(newQueued[txNonce], newTx)
-          } else {
-            newQueued = { ...newQueued, [txNonce]: [newTx] }
-          }
-        } else {
-          if (newNext?.[txNonce]) {
-            newNext[txNonce] = getUpdatedQueueTxs(newNext[txNonce], newTx)
-          } else {
-            newNext = { [txNonce]: [newTx] }
-          }
-          // Remove queued transaction from queue that moved to next
-          if (newQueued?.[txNonce]) {
-            delete newQueued[txNonce]
-          }
-        }
-      })
 
-      // No new txs, empty queue list, cleanup
-      if (!values.length && !Object.keys(newQueued).length && Object.keys(newNext).length === 1) {
-        newNext = {}
+      // From the list of TransactionItems, create two sub-lists split by where the "Queued" label is.
+      // If there's no "Queued" label, put all the items into the "next" group.
+      let nextItems = values
+      let queuedItems = values.slice(0, 0)
+      const qLabelIndex = values.findIndex((item) => isLabel(item) && item.label === LabelValue.Queued)
+      if (qLabelIndex > -1) {
```

Comment:
For consistency with the history reducer, I'd check against `0` (or change it above).
```suggestion
      if (qLabelIndex >= 0) {
```

Reply:
Adjusted the other two places.